hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/DistributionBarHelper.java

```diff
@@ -0,0 +1,134 @@
+    }
+
+    private static String getPartStyle(int partIndex, int noOfParts, String customStyle) {
+        StringBuilder mainStyle = new StringBuilder();
+        StringBuilder styleName = new StringBuilder(GwtDistributionBar.CLASSNAME);
+        if (noOfParts == 1) {
+            styleName.append("-only");
+        } else if (partIndex == 1) {
+            styleName.append("-left");
+        } else if (partIndex == noOfParts) {
+            styleName.append("-right");
+        } else {
+            styleName.append("-middle");
+        }
+        mainStyle.append(styleName).append(" ");
+        mainStyle.append(DISTRIBUTION_BAR_PART_MAIN_STYLE).append(" ");
+        mainStyle.append(DISTRIBUTION_BAR_PART_CLASSNAME_PREFIX + partIndex);
+        if (customStyle != null) {
+            mainStyle.append(" ").append("status-bar-part-" + customStyle);
+        }
+        return mainStyle.toString();
+    }
+
+    private static String getPartWidth(Long value, Long totalValue, int noOfParts) {
+        final double minTotalSize = MINIMUM_PART_SIZE * noOfParts;
+        final double availableSize = PARENT_SIZE_IN_PCT - minTotalSize;
+        double val = MINIMUM_PART_SIZE + (double) value / totalValue * availableSize;
+        return String.format("%.3f", val) + "%";
+    }
+
+    private static String getPart(int partIndex, Status status, Long value, Long totalValue, int noOfParts) {
+        String partValue = status.toString().toLowerCase();
+        // return "<div class=\"" + getPartStyle(partIndex, noOfParts,
+        // partValue) + "\" style=\"width: "
+        // + getPartWidth(value, totalValue, noOfParts) + ";\" title = \"" +
+        // partValue + "\"><span class=\""
+        // + DISTRIBUTION_BAR_PART_VALUE_CLASSNAME + "\">" + value +
+        // "</span></div>";
+        return "<div class=\"" + getPartStyle(partIndex, noOfParts, partValue) + "\" style=\"width: "
+                + getPartWidth(value, totalValue, noOfParts) + ";\"><span class=\""
+                + DISTRIBUTION_BAR_PART_VALUE_CLASSNAME + "\">" + value + "</span></div>";
+    }
+
+    public static String getDistributionBarAsHTMLString(Map<Status, Long> statusTotalCountMap) {
+        StringBuilder htmlString = new StringBuilder();
+        htmlString.append(getParentDivStart());
+        Long totalValue = getTotalSizes(statusTotalCountMap);
+        Map<Status, Long> statusMapWithNonZeroValues = getNonZeroStatusList(statusTotalCountMap);
+
+        if (statusMapWithNonZeroValues.size() > 0) {
+            int partIndex = 1;
+            for (Map.Entry<Status, Long> entry : statusMapWithNonZeroValues.entrySet()) {
+                if (entry.getValue() > 0) {
+                    htmlString.append(getPart(partIndex, entry.getKey(), entry.getValue(), totalValue,
+                            statusMapWithNonZeroValues.size()));
+                    partIndex++;
+                }
+            }
+        } else {
+            return getUnintialisedBar();
```

Comment:
readability: return statement in the "true" block, then is no else necessary
 if (statusMapWithNonZeroValues.size() <= 0) {
  return getUnintialisedBar();
}
...